src/execution/__tests__/subscribe-test.ts

```diff
@@ -152,9 +151,9 @@ const DummyQueryType = new GraphQLObjectType({
   },
 });
 
-async function subscribeWithBadFn(
+function subscribeWithBadFn(
   subscribeFn: () => unknown,
-): Promise<ExecutionResult> {
+): PromiseOrValue<ExecutionResult> {
```

Comment:
We still have `await subscribeWithBadFn` in a few places.
It was a non-issue when we always returned promises, but now it means we don't check if it's a promise before resolving it.

Reply:
to clarify I'm speaking about "await" working with both promises and non-promises.

Reply:
As a temporary, solution you can do the same approach of saving to result and checking it with `isPromise`.

Reply:
fixed awaits, and used `expectPromise.toResolve()` to ensure is promise where appropriate. [f88b523](https://github.com/graphql/graphql-js/pull/3620/commits/f88b523c175b1c1280dc4659be12426f8f9ea07f)